src/marketlist.schema.json

```diff
@@ -106,13 +106,12 @@
       "description": "Parameters of the contract containing a `divest` method to exit an investment opportunity",
       "properties": {
         "address": {
+          "description": "Address of the contract containing the invest logic",
+          "pattern": "^0x[a-fA-F0-9]{40}$",
           "type": "string"
         },
         "signature": {
-          "enum": [
-            "function divest(address recipient, uint256[] calldata receiptAmounts, uint256[] calldata rewardAmounts) external payable",
-            "function divest(address recipient, uint256[] calldata receiptAmounts, uint256[] calldata rewardAmounts, bytes calldata data) external payable"
-          ],
+          "pattern": "^function\\sdivest((s*([a-zA-Z_]w+)s+([a-zA-Z_]w+),?)*)",
```

Comment:
This is an 80/20 version of function validation.

It validates that the signature starts with `function divest` and then has any number of arguments.

But, it doesn't validate the post parameter terms (ie: external, payable).  Didn't seem worth it spend more time on those.

Reply:
It seems this only matches the `function divest` part of the string (as expected), so what is all the stuff after `\sdivest` for? Seems we can remove that to and do this

```suggestion
          "pattern": "^function\\sdivest\\(.*\\).*(external|public)",
```

![image](https://user-images.githubusercontent.com/17163988/157134410-170831e5-865f-494a-a365-ee2ed1affa7c.png)


![image](https://user-images.githubusercontent.com/17163988/157134234-057584ed-cdbe-448d-be14-ed386158934d.png)

Reply:
The possibilities are the combinations of each bullet point below (so there's a lot of options):
- must have `external` or `public` (we only use `external`, but `public` is equally valid)
- will be either `payable` or not payable, in which case it just won't say `payable`
- might be `virtual`
- might `override` a method

So I think that gives 16 possibilities

Reply:
Got it.   Ok, in that case, let's just keep it simple and go with your suggestion that only forces the presence of external or public.